File: skypydb/utils/embedding_callable_mixin.py
"""Mixin that allows embedding providers to be used as callables."""

from typing import List


class EmbeddingCallableMixin:
    def __call__(
        self,
        texts: List[str]
    ) -> List[List[float]]:
        """Make embedding providers callable for compatibility with integrations."""

        return self.embed(texts)

File: skypydb-js/src/embeddings/mixins/embedding_callable_mixin.ts
import type { EmbeddingMatrix } from "../../types";

export abstract class EmbeddingCallableMixin {
  abstract embed(texts: string[]): Promise<EmbeddingMatrix>;

  async call(texts: string[]): Promise<EmbeddingMatrix> {
    return this.embed(texts);
  }
}
